src/Bicep.Core/Semantics/ArmTemplateSemanticModel.cs

```diff
@@ -215,6 +218,31 @@ private TypeSymbol GetType(TemplateOutputParameter output)
                     .Select(kvp => new ExportedTypeMetadata(kvp.Key, GetType(kvp.Value), GetMostSpecificDescription(kvp.Value))));
             }
 
+            if (template.Functions is { } userDefinedFunctions)
+            {
+                foreach (var @namespace in userDefinedFunctions)
+                {
+                    // User-defined functions in ARM *must* be namespaced -- a missing namespace name indicates an invalid or incomplete template.
+                    if (@namespace.Namespace?.Value is not string nsName)
+                    {
+                        continue;
+                    }
+
+                    // A user-defined function residing in the namespace used by the Bicep compiler must be imported by its unqualified name.
+                    // Functions in any other namespace must be imported using their fully qualified name ("<namespace name>.<function name>")
+                    var namePrefix = !nsName.Equals(EmitConstants.UserDefinedFunctionsNamespace) ? $"{nsName}." : string.Empty;
+
+                    exports.AddRange(@namespace.Members.Where(kvp => IsExported(kvp.Value))
+                        .Select(kvp => new ExportedFunctionMetadata(
+                            Name: $"{namePrefix}{kvp.Key}",
+                            Parameters: kvp.Value.Parameters is TemplateFunctionParameter[] parameters
```

Comment:
[nit] can shorten to `kvp.Value.Parameters is {} parameters`, if you're just trying to check non-null.